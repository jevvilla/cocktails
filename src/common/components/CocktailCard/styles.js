import { StyleSheet } from 'react-native';

import colors from '../../colors';
import styles from '../../styles';

export default StyleSheet.create({
  container: {
    ...styles.borderRadius,
    ...styles.padding,
    backgroundColor: colors.white,
    flexDirection: 'row',
    height: 180,
    justifyContent: 'space-between',
  },
  descriptionWrapper: {
    ...styles.paddingRight,
    flex: 1,
  },
  image: {
    ...styles.borderRadius,
    width: 150,
  },
  touchable: {
    ...styles.borderRadius,
    ...styles.marginBottom,
    ...styles.shadow(4),
  },
  title: {
    ...styles.marginBottom,
    color: colors.black,
    fontSize: 22,
    opacity: 0.6,
  },
  ingredient: {
    color: colors.black,
    fontSize: 16,
    opacity: 0.4,
  },
});
