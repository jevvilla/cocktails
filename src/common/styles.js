import colors from './colors';

export const STANDARD_SPACING = 16;
export const STANDARD_RADIUS = 8;

export const DOUBLE_SPACING = STANDARD_SPACING * 2;

export default {
  /* Paddings */
  padding: {
    padding: STANDARD_SPACING,
  },

  paddingHorizontal: {
    paddingHorizontal: STANDARD_SPACING,
  },

  paddingVertical: {
    paddingVertical: STANDARD_SPACING,
  },

  paddingRightHalf: {
    paddingRight: STANDARD_SPACING,
  },

  /* Margins */
  marginBottom: {
    marginBottom: STANDARD_SPACING,
  },

  /* Border Radius */
  borderRadius: {
    borderRadius: STANDARD_RADIUS,
  },

  /* Shadow */
  shadow: value => ({
    elevation: value,
    shadowOpacity: 0.2,
    shadowRadius: value,
    shadowOffset: { width: 0, height: 0 },
  }),

  /* Header */
  defaultHeader: {
    borderBottomWidth: 0,
    backgroundColor: colors.background,
    elevation: 0,
    height: 64,
  },

  headerTitle: {
    color: colors.white,
    fontSize: 22,
    opacity: 0.8,
  },
};
