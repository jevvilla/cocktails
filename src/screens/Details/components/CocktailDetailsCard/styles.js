import { StyleSheet } from 'react-native';

import colors from '../../../../common/colors';
import styles from '../../../../common/styles';

export default StyleSheet.create({
  container: {
    ...styles.borderRadius,
    ...styles.padding,
    backgroundColor: colors.white,
    height: 650,
  },
  imageContainer: {
    flex: 1.3,
  },
  descriptionContainer: {
    ...styles.paddingVertical,
    flex: 1,
  },
  image: {
    flex: 1,
  },
  measures: {
    color: colors.black,
    fontSize: 16,
    opacity: 0.4,
  },
  howTo: {
    ...styles.paddingVertical,
  },
});
