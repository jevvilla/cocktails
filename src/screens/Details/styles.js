import { StyleSheet } from 'react-native';

import colors from '../../common/colors';
import styles from '../../common/styles';

export default StyleSheet.create({
  container: {
    ...styles.paddingVertical,
    ...styles.paddingHorizontal,
    backgroundColor: colors.background,
    flex: 1,
  },
  headerLeft: {
    ...styles.paddingHorizontal,
  },
  headerTitle: {
    ...styles.headerTitle,
  },
});
