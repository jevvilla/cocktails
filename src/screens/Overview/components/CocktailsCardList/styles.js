import { StyleSheet } from 'react-native';

import styles from '../../../../common/styles';

export default StyleSheet.create({
  flatListContainer: {
    ...styles.paddingHorizontal,
    ...styles.paddingVertical,
  },
});
