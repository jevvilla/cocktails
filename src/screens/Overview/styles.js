import { StyleSheet } from 'react-native';

import colors from '../../common/colors';
import styles from '../../common/styles';

export default StyleSheet.create({
  container: {
    backgroundColor: colors.background,
    flex: 1,
  },
  headerTitle: {
    ...styles.headerTitle,
  },
});
