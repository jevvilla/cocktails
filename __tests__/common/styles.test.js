import styles, { STANDARD_SPACING } from '../../src/common/styles';

describe('styles', () => {
  test('should match snapshot', () => {
    expect(styles).toMatchSnapshot();
  });

  test('STANDARD_SPACING should match snapshot', () => {
    expect(STANDARD_SPACING).toMatchSnapshot();
  });
});
