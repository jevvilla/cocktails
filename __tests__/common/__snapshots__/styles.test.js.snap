// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles STANDARD_SPACING should match snapshot 1`] = `16`;

exports[`styles should match snapshot 1`] = `
Object {
  "borderRadius": Object {
    "borderRadius": 8,
  },
  "defaultHeader": Object {
    "backgroundColor": "#41AFC7",
    "borderBottomWidth": 0,
    "elevation": 0,
    "height": 64,
  },
  "headerTitle": Object {
    "color": "#fff",
    "fontSize": 22,
    "opacity": 0.8,
  },
  "marginBottom": Object {
    "marginBottom": 16,
  },
  "padding": Object {
    "padding": 16,
  },
  "paddingHorizontal": Object {
    "paddingHorizontal": 16,
  },
  "paddingRightHalf": Object {
    "paddingRight": 16,
  },
  "paddingVertical": Object {
    "paddingVertical": 16,
  },
  "shadow": [Function],
}
`;
